internal/pkg/aws/session/session.go

```diff
@@ -108,3 +116,25 @@ func (p *Provider) FromRole(roleARN string, region string) (*session.Session, er
 	sess.Handlers.Build.PushBackNamed(userAgentHandler())
 	return sess, nil
 }
+
+// ProfileNames returns a list of profile names available in the user's ~/.aws/config file.
+// An error is returned if the config file can't be found or parsed.
+func (p *Provider) ProfileNames() ([]string, error) {
```

Comment:
Why do we want "ProfileNames" to belong to "Provider"🤔since it does not make use of "Provider". Would it be better to have a `profile` package in "aws" dir and move content of "ini" package in it as well?

Reply:
yeah to be fair "provider" doesn't store anything useful, it's only used for us to be able to mock session creation :D 

I don't mind the `profile` package idea! I can do that. I put it in `session` because it's kindof couple always with `session.FromProfile` afterwards

The `ini` package should be still separate because it's not a unique format to AWS profiles, it can be used by other packages.

Reply:
Done!